shared/django_apps/rollouts/models.py

```diff
@@ -0,0 +1,50 @@
+from random import choice
+
+from django.contrib.postgres import fields
+from django.db import models
+
+
+def default_random_salt():
+    chars = []
+    ALPHABET = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    for _ in range(16):
+        chars.append(choice(ALPHABET))
+    return "".join(chars)
+
+
+class FeatureFlag(models.Model):
+    name = models.CharField(max_length=200, primary_key=True)
+    proportion = models.DecimalField(default=0, decimal_places=3, max_digits=3)
+    salt = models.CharField(max_length=10000, default=default_random_salt)
+
+    class Meta:
+        db_table = "feature_flags"
+        constraints = [
+            models.UniqueConstraint(fields=["name"], name="feature_flag_name")
+        ]
+        # TODO: assert that variant proportions sum to 1
+
+
+class FeatureFlagVariant(models.Model):
+    name = models.CharField(max_length=200, primary_key=True)
+    feature_flag = models.ForeignKey(
+        "FeatureFlag", on_delete=models.CASCADE, related_name="variants"
+    )
+    proportion = models.DecimalField(default=0, decimal_places=3, max_digits=3)
+    enabled = models.BooleanField(default=False)
+
+    # Weak foreign keys to Owner and Respository models respectively
+    override_owner_ids = fields.ArrayField(
+        base_field=models.IntegerField(), default=list
+    )
+    override_repo_ids = fields.ArrayField(
+        base_field=models.IntegerField(), default=list
+    )
+
+    # TODO: maybe add more fields for more granularity on feature variants. EG: featureA uses value 10 vs featureB uses value 100
```

Comment:
idk if this would be useful, for now probably not

Reply:
i don't think we need to reuse a FeatureVariant for multiple Features. the values and overrides are all feature-specific so we may as well make the variants feature-specific as well